docs/TestingHighLevel.md

```diff
@@ -1,31 +1,62 @@
 <!--- SPDX-License-Identifier: Apache-2.0 -->
 
-# ONNX-MLIR: Testing and Specific Environment variables
+# ONNX-MLIR: Build trouble-shooting and testing ONNX_MLIR
+
+## Trouble shooting the building of ONNX-MLIR
+
+If you have issues during the first `onnx-mlir` build, you may need to check the environment variables used by our build. See the last section of this page for help.
+
+If you have used the source directory successfully for a while, you may experience difficulties to rebuild `onnx-mlir` after merging the latest changes from the `main` branch.
+
+Below is a couple of steps you may perform. If any of them apply, my recommendation is to remove the `onnx-mlir/build` subdirectory and rebuild from scratch using the `cmake` commands.
+
+### 1) Checking the right commit of the llvm-project
+
+If the latest `onnx-mlir` `main` branch has moved to a newer commit level of the `llvm-project`, the build process will typically experience multiple compiler failures related to LLVM and MLIR code.
+
+Level required is found in the first code box of this page: (Building ONNX-MLIR)[BuildOnLinuxOSX.md#mlir].
+
+Level used in the code is found by executing a `git log` in the `llvm-project` subdirectory.
+
+If they don't match, please update the llvm project to the required level.
+
+### 2) Checking the right third_party support
+
+Typically, when we update the ONNX op level, it results in new software in the `third_party/onnx` subdirectory. Failing to update that code results typically in compiler failures related to ONNX dialect code.
+
+I find it easier to simply remove the `third_party` directory and then reinstalling the code using `git submodule update --init --recursive`.
```

Comment:
[suggestion]: "I find it easier" -> "It is easier"

Reply:
yes, tx